src/script/op_codes.rs

```diff
@@ -36,92 +38,301 @@ pub fn to_boxed_hash(slice: &[u8]) -> Box<HashValue> {
 pub const OP_PUSH_ZERO: u8 = 0x7b;
+pub const OP_NOP: u8 = 0x73;
+pub const OP_PUSH_ONE: u8 = 0x7c;
+pub const OP_PUSH_INT: u8 = 0x7d;
+pub const OP_PUSH_PUBKEY: u8 = 0x7e;
 
-// Opcode constants: Comparisons
+// Opcode constants: Math Operations
 pub const OP_EQUAL: u8 = 0x80;
 pub const OP_EQUAL_VERIFY: u8 = 0x81;
-
-// Opcode constants: Arithmetic
 pub const OP_ADD: u8 = 0x93;
 pub const OP_SUB: u8 = 0x94;
+pub const OP_GE_ZERO: u8 = 0x82;
+pub const OP_GT_ZERO: u8 = 0x83;
+pub const OP_LE_ZERO: u8 = 0x84;
+pub const OP_LT_ZERO: u8 = 0x85;
 
-// Opcode constants: Cryptography
+// Opcode constants: Boolean Logic
+pub const OP_OR_VERIFY: u8 = 0x64;
+pub const OP_OR: u8 = 0x65;
+
+// Opcode constants: Cryptographic Operations
 pub const OP_CHECK_SIG: u8 = 0xac;
 pub const OP_CHECK_SIG_VERIFY: u8 = 0xad;
 pub const OP_HASH_BLAKE256: u8 = 0xb0;
+pub const OP_HASH_SHA256: u8 = 0xb1;
+pub const OP_HASH_SHA3: u8 = 0xb2;
+
+// Opcode constants: Miscellaneous
+pub const OP_RETURN: u8 = 0x60;
+pub const OP_IF_THEN: u8 = 0x61;
+pub const OP_ELSE: u8 = 0x62;
+pub const OP_END_IF: u8 = 0x63;
 
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub enum Opcode {
-    /// Push the current chain height onto the stack
-    PushHeight,
-    /// Push the associated 32-byte value onto the stack
-    PushHash(Box<HashValue>),
-    /// Push a zero onto the stack
+    // Block Height Checks
+    /// Compare the current block height to height. Fails with VERIFY_FAILED if the block height < height.
+    CheckHeightVerify(u64),
+    /// Pushes the value of (the current height - height) to the stack. In other words, the top of the stack will hold
+    /// the height difference between height and the current height. If the chain has progressed beyond height, the
+    /// value is positive; and negative if the chain has yet to reach height. Fails with STACK_OVERFLOW if the stack
+    /// would exceed the max stack height.
+    CheckHeight(u64),
+    /// Pops the top of the stack as height and compares it to the current block height. Fails with INVALID_INPUT
+    /// if there is not a valid integer value on top of the stack. Fails with EMPTY_STACK if the stack is empty.
+    /// Fails with VERIFY_FAILED if the block height < height.
+    CompareHeightVerify,
+    /// Pops the top of the stack as height, then pushes the value of (height - the current height) to the stack. In
+    /// other words, this opcode replaces the top of the stack with the difference between that value and the current
+    /// height. Fails with INVALID_INPUT if there is not a valid integer value on top of the stack. Fails with
+    /// EMPTY_STACK if the stack is empty.
+    CompareHeight,
+
+    // Stack Manipulation
+    /// No op. Does nothing. Never fails.
+    Nop,
+    /// Pushes a zero onto the stack. This is a very common opcode and has the same effect as PushInt(0) but is more
+    /// compact. PushZero can also be interpreted as PushFalse (although no such opcode exists). Fails with
```

Comment:
```suggestion
    /// compact. Fails with
```
I think we can remove this, as this is part of the previous idea where 1 is script = true, and 0 is script = fail.